plugins/cilium-cni/cilium-cni.go

```diff
@@ -324,6 +325,48 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}
 	defer netNs.Close()
 
+	if len(n.NetConf.RawPrevResult) != 0 {
+		err := cniVersion.ParsePrevResult(&n.NetConf)
+		if err != nil {
+			return fmt.Errorf("unable to understand network config: %s", err)
+		}
+		str := strings.Split(args.Netns, "/")
+		pid, err := strconv.Atoi(str[2])
+		if err != nil {
+			return err
+		}
+		v6, err := c.IPAMAllocate("ipv6")
+		if err != nil {
+			logger.WithError(err).Warn("unable to allocate IPv6 address")
+			return fmt.Errorf("unable to allocate IPv6 address: %s", err)
+		}
+		defer func() {
+			if err != nil {
+				c.IPAMReleaseIP(v6.Address.IPV6)
+			}
+		}()
+
+		ep, err := connector.DeriveEndpointFrom("cni0", args.ContainerID, pid)
+		if err != nil {
+			logger.WithError(err).WithFields(logrus.Fields{
+				logfields.ContainerID: args.ContainerID}).Warn("Unable to derive endpoint")
+			return fmt.Errorf("unable to derive endpoint: %s", err)
```

Comment:
>I assume that if someone misconfigures their system to unintentionally have two CNIs, and they're not running flannel, then this is the path it will hit?

That can never happen because we are the ones that setup the configuration and so far the only configuration available is the `flannel` chained with `cilium`, there should be `04-cilium.conflist`  somewhere in this PR

Reply:
Ah, so we generate the CNI config file based upon whether the user configured this option in the Cilium DS? That answers my questions entirely. Thanks.